### PR TITLE
Add recipe for `r-spgwr`

### PR DESCRIPTION
**Pull request by darribas**
This PR adds a recipe for `r-spgwr`, built for the [`spgwr`](https://cran.r-project.org/web/packages/spgwr/index.html) R package. It has been built using [`conda_r_skeleton_helper`](https://github.com/bgruening/conda_r_skeleton_helper).

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example). (**DAB NOTE**: there is a license set by the package but it does not include a file for it).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged. (**DAB NOTE**: unsure, not checking)
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md). (**DAB NOTE**: unsure, not checking)
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there. (**DAB NOTE**: @darribas is willing to be listed here)
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

I'm cc'ing @rsbivand as he might have further insight into the license issues I've not been able to clarify above.

Also paging @mfansler as this is my first PR and he suggested he'd be willing to help with the process.

@conda-forge-admin, please ping team (I _think_ I need to do this, otherwise, paging @conda-forge/help-r)

### CODE REVIEW

**Comment by darribas** [comment]
I'm willing to be listed on the maintainer list

**Comment by rsbivand** [comment]
@darribas: `spgwr` almost got retired last year, and really should not be used, it is not actively maintained. 

`GWmodel` is a bit more actively maintained. 

Beyond that, I never use and recommend that `conda` never be use with R. Anaconda is not a community participant and was unresponsive when approached while I was working on retiring `rgdal` etc. last year. 

I would be happy to ceded maintainership of `spgwr` at the earliest opportunity, provided that the start-up message is kept (asking users to avoid GWR for anything beyond ESDA).

**Comment by ocefpaf** [comment]
> Beyond that, I never use and recommend that `conda` never be use with R. Anaconda is not a community participant and was unresponsive when approached while I was working on retiring `rgdal` etc. last year.

The conda-forge is not Anaconda. The conda package manager and the conda community are not Anaconda.

**Comment by mfansler** [comment]
@rsbivand RE: r-rgdal please post an issue on [the Conda Forge feedstock](https://github.com/conda-forge/r-rgdal-feedstock) with your recommendations. Or feel free to reach out to me directly - I'm always happy to maintain open communication with upstream developers.

**Comment by mfansler** [comment]
@darribas R is distributed with a set of common licenses. CRAN allows packages to ship without an explicit license text if it uses one of those. What we do on Conda Forge is have the individual package make a copy of the license(s) found in the `r-base` package. The `conda_r_skeleton_helper` already did this for you:

```
  license_file:
     - '{{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2'
```

However, since the actual package has `GPL (>= 2)`, I usually manually expand this to

```
  license_file:
     - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2
     - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-3
```

which keeps parity with how CRAN expands it on the website.

The linter, however, is making a separate complaint about SPDX formatting, namely that `license: GPL (>= 2)` needs to be `GPL-2.0-or-later`.

File: recipes/r-spgwr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+requirements:
+  build:
+    - {{ compiler('c') }}              # [not win]
+    - {{ compiler('m2w64_c') }}        # [win]
```
Comment (mfansler) [inline recipes/r-spgwr/meta.yaml:25]: Needs `stdlib` declarations:

```suggestion
    - {{ compiler('c') }}        # [not win]
    - {{ stdlib('c') }}          # [not win]
    - {{ compiler('m2w64_c') }}  # [not win]
    - {{ stdlib('m2w64_c') }}    # [not win]
```

File: recipes/r-spgwr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    - r-sp >=0.8_3
+    - r-spdata >=2.3.1
+
+test:
+  commands:
+    - $R -e "library('spgwr')"           # [not win]
+    - "\"%R%\" -e \"library('spgwr')\""  # [win]
+
+about:
+  home: https://github.com/rsbivand/spgwr/, https://rsbivand.github.io/spgwr/
```
Comment (mfansler) [inline recipes/r-spgwr/meta.yaml:48]: Split URLs:

```suggestion
  home: https://rsbivand.github.io/spgwr/
  dev_url: https://github.com/rsbivand/spgwr/
```

File: recipes/r-spgwr/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  license: GPL (>= 2)
```
Comment (mfansler) [inline recipes/r-spgwr/meta.yaml:49]: SPDX:

```suggestion
  license: GPL-2.0-or-later
```

File: recipes/r-spgwr/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  summary: Functions for computing geographically weighted regressions are provided, based on
+    work by Chris Brunsdon, Martin Charlton and Stewart Fotheringham.
+  license_family: GPL
+  license_file:
+    - '{{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2'
```
Comment (mfansler) [inline recipes/r-spgwr/meta.yaml:54]: Also include v3:

```suggestion
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-3
```

File: recipes/r-spgwr/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    - {{ cran_mirror }}/src/contrib/Archive/spgwr/spgwr_{{ version }}.tar.gz
+  sha256: d357d84b957ab5808230bb13fa65beebef730fd4d703ab5604307218fd22b4d3
+
+build:
+  merge_build_host: True  # [win]
+  number: 0
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
```
Comment (mfansler) [inline recipes/r-spgwr/meta.yaml:21]: We can skip windows for now and the migrator can retry it on the feedstock.

```suggestion
  skip: true  # [win]

```

**Review by mfansler — CHANGES_REQUESTED** [review]
The technical side of recipe has a few things to touch up, but mostly fine. However, given @rsbivand comments, perhaps you don't require this package after all. Obviously, this depends on your needs. If you are rerunning older code, perhaps you will still want this; if not, it may be prudent to pre-emptively choose not distribute this here if a dominantly better alternative is already available.

If you want to proceed with adding this here, then the noted changes should be made. Otherwise, you can choose to close.

**Comment by darribas** [comment]
@mfansler Thank you so much for the detailed and careful review. I think, based on the discussion above, I will not add `r-spgwr` to conda-forge and instead explore alternatives (e.g., `GWmodel`). These edits are super useful though, I'll revisit them when creating other packages. Closing for now on that basis.

**Pull request closed without merging**